AppController/lib/cron_helper.rb

```diff
@@ -288,31 +288,133 @@ def self.convert_messy_format(schedule)
     cron = ""
     splitted = schedule.split
 
-    unless splitted.length == 3 or splitted.length == 5
+    unless splitted.length == 3 || splitted.length == 5 || splitted.length == 7
       Djinn.log_error("bad format, length = #{splitted.length}")
       return [""]
     end
 
     ord = splitted[0]
     days_of_week = splitted[1]
-    months_of_year = ""
-    time = ""
+
+    multiple_cron_entries = false
+    crons = Array.new
 
     if splitted.length == 3
       months_of_year = "every"
       time = splitted[2]
-    else
+      hour, min = time.split(":")
+    elsif splitted.length == 5
       months_of_year = splitted[3]
       time = splitted[4]
+      hour, min = time.split(":")
+    else    # schedule length = 7, e.g. every 7 minutes from 10:00 to 14:00
+      months_of_year = "every"
+      days_of_week = "day"
+
+      increment = splitted[1]
+      increment_type = splitted[2]
+
+      # Split hour and minute and trim leading zeros.
+      # Start time is represented as t1 = h1:m1.
+      # End time is represented as t2 = h2:m2.
+      # The string values are used for the cron representation
+      # and the integer values are used for comparisons.
+      h1, m1 = splitted[4].split(":").map(&:to_i).map(&:to_s)
```

Comment:
I think it would be more readable if you kept these as integers and then used the the string format notation (eg. `"#{h1}-#{h2 - 1}/#{increment}"` below.